### PR TITLE
feat: track upload

### PR DESCRIPTION
**Pull request by n0izn0iz**
Tracks are Posts with special Metadata
image

image

image

Social Feed
image

UPP
image

### CODE REVIEW

File: packages/components/socialFeed/NewsFeed/NewsFeedQueries.ts [hunk 1]
```diff
@@ -256,3 +258,12 @@ export const generatePostMetadata = ({
     updatedAt: new Date().toISOString(),
   };
 };
+
+export const isValidMusicAudioPost = (post: Post) =>
+  post.category === PostCategory.MusicAudio &&
+  ZodTrack.safeParse(JSON.parse(post.metadata)).success;
```
Comment (n0izn0iz) [inline packages/components/socialFeed/NewsFeed/NewsFeedQueries.ts:264]: this function can throw if JSON.parse fails, maybe wrap it in a try catch

File: packages/components/socialFeed/NewsFeed/NewsFeedQueries.ts [hunk 2]
```diff
@@ -256,3 +258,12 @@ export const generatePostMetadata = ({
...
+
+export const getMusicAudioPostTrack = (post: Post): Track | undefined =>
+  isValidMusicAudioPost(post)
+    ? ZodTrack.parse(JSON.parse(post.metadata))
```
Comment (n0izn0iz) [inline packages/components/socialFeed/NewsFeed/NewsFeedQueries.ts:268]: same

File: packages/components/socialFeed/NewsFeed/NewsFeedQueries.ts (hunk 1 above)
Comment (WaDadidou) [reply packages/components/socialFeed/NewsFeed/NewsFeedQueries.ts:264]: Oki, done

File: packages/components/socialFeed/NewsFeed/NewsFeedQueries.ts (hunk 2 above)
Comment (WaDadidou) [reply packages/components/socialFeed/NewsFeed/NewsFeedQueries.ts:268]: Oki, done

**Comment by WaDadidou** [comment]
I don't understand why my uploaded tracks doesn't appear in UPP, with this `Posts` request :cry:
It's ok with `user: ""`, but not with `user: userId`
But a filter with something like `user: userId` + `categories: [PostCategory.Audio]` works

image
image
image

**Comment by n0izn0iz** [comment]
it was a backend issue, resolved

**Review by omniwired — DISMISSED** [review]

**Pull request merged**